image/templates/helm/shared/templates/_openshift.tpl

```diff
@@ -45,3 +45,41 @@
 {{ end }}
 
 {{ end }}
+
+{{/*
+    srox.initOpenshiftMonitoring $
+
+    This function enables the Openshift Monitoring variables and flags.
+*/}}
+{{ define "srox.initOpenshiftMonitoring" }}
+
+{{ $ := index . 0 }}
+
+{{ if $._rox.enableOpenShiftMonitoring }}
+  {{ include "srox.warn" (list . "enableOpenShiftMonitoring option was replaced with monitoring.openshift.enabled") }}
+  {{ $_ := set $._rox "monitoring" (dict "openshift" (dict "enabled" true)) }}
+{{ end }}
+
+{{/* Default `monitoring.openshift.enabled = true` unless `env.openshift != 4`. */}}
+{{ if kindIs "invalid" $._rox.monitoring.openshift.enabled }}
+  {{ $_ := set $._rox "monitoring" (dict "openshift" (dict "enabled" (eq $._rox.env.openshift 4))) }}
+{{ end }}
+
+{{ if and $._rox.monitoring.openshift.enabled (ne $._rox.env.openshift 4) }}
+  {{ include "srox.warn" (list . "'monitoring.openshift.enabled' is set to true, but the chart is not being deployed in an OpenShift 4 cluster. Proceeding with 'monitoring.openshift.enabled=false'.") }}
+  {{ $_ := set $._rox "monitoring" (dict "openshift" (dict "enabled" false)) }}
+{{ end }}
+
+{{/*
+  Detect if the secured-cluster-services chart is deployed in the same namespace
```

Comment:
What if the secured cluster chart is installed first? I think it might be easier to use dedicated roles for each chart to avoid coupling.

Reply:
> What if the secured cluster chart is installed first?

Do we support installing Central + Secured in a namespace while Secured is connected to a separate Central instance in another namespace? I am thinking Secured needs certs from Central first, so either it has one from a separate Central or Central in the same namespace needs to be deployed first.

Reply:
> Do we support installing Central + Secured in a namespace while Secured is connected to a separate Central instance in another namespace?

I think technically we do support that, or at least let's say it's not unsupported. Also I think you can create a secured cluster CR first and then apply the secret from the init bundle afterwards.

Anyways, I think we should probably avoid sharing resources between different Helm charts. There is also the issue of cleaning up the resources - we should be able to remove and re-install both Central and Secured cluster charts independently from each other.

Reply:
@stehessel The same logic to [determine if the secured cluster is being deployed with Central](https://github.com/stackrox/stackrox/blob/jvdm%2Fmetrics%2Fsecure-metrics/image/templates/helm/shared/templates/_openshift.tpl#L73-L83) is used to determine if the [Scanner V4 Indexer should be deployed n the Secured Cluster or not](https://github.com/stackrox/stackrox/blob/jvdm%2Fmetrics%2Fsecure-metrics/image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl#L138-L147)--that is, we don't install the Secured Cluster's Scanner V4 Indexer and assume Central's chart will provide it. The same would apply to the reader Cluster Role. If we detect Central, we do not create the role but assume Central's chart will provide it.

It doesn't look like we support deploying Scanner V4 by doing Secured Cluster and then Central. You gotta a point about breaking the ability to deploy it that way if we do the same with the Cluster Role and, let's say, Scanner V4 is disabled. Scanner V4 is planned to become on by default, so I am not sure if what you want to support aligns with the bigger picture.

Can you provide more pointers? Is this a scenario we want to support?

BTW, I can always go back to using namespaced roles as well -- since supporting the ClusterRole is outside the original goal of this PR. I am doing it to see if I can help out with that TODO item, but let me know.

Reply:
Hmm I did not realize that we already established a strong coupling between the central and secured cluster charts. I don't think this is a good idea and I can easily some problems caused by the causality for use cases where the installation order is not guaranteed. I'm thinking about GitOps setups like ArgoCD for example. Anyways, I guess the damage is done.

For our situation, what do you think about the following:

* We let the central and secured cluster charts each manage their own `ClusterRole` and `ClusterRoleBinding` (`central-...` / `secured-cluster-...`). We put a namespace suffix to make the names unique per namespace.
* If scanner V4 is enabled for the Central chart or the secured cluster chart, we put the scanner v4 service account into the respective `ClusterRoleBinding`.
* If both charts are installed into the same namespace, the scanner v4 service account would be bound to both cluster roles, but this shouldn't cause any problems.

This way we avoid any coupling / causality requirements and we don't have to perform a Helm `lookup`. The cost is a potential duplication of the cluster role and binding.

Reply:
Are we working around some limitations on how Helm + RBAC was originally designed to be used?

Helm was not designed to manage global resources across charts. And RBAC doesn't have a mechanism to bind on roles from outside the namespace. Cluster Roles were designed to be one resource for cross-namespace authorization. We are abusing Cluster Roles to allow us to create per-namespace bindings on namespace Roles.

It's OK to try to reduce the coupling between the two charts, but this effort will not bring much benefit in the long term unless we also discuss bigger coupling points, such as Scanner V4 deployments. And I am not fully familiar with these charts, so I might not be aware of other coupling points.

Reply:
@stehessel I am reverting this to using the current approach, adding bindings to `kube-system`. For the reasons above and for the fact that following up on the questions above is orthogonal to this PR goal. I am resolving the related threads.

Reply:
> It's OK to try to reduce the coupling between the two charts, but this effort will not bring much benefit in the long term unless we also discuss bigger coupling points, such as Scanner V4 deployments. And I am not fully familiar with these charts, so I might not be aware of other coupling points.

Agreed, I guess it's also a bit futile trying to solve this between the two of us when the scanner v4 deployment was actually done by Draconians.